apps/onboarding-functions/src/main/java/it/pagopa/selfcare/onboarding/service/CompletionServiceDefault.java

```diff
@@ -180,6 +193,22 @@ public void sendCompletedEmail(Onboarding onboarding) {
                 destinationMails, product, onboarding.getInstitution().getInstitutionType());
     }
 
+    @Override
+    public void persistUsers(Onboarding onboarding) {
+        List<User> users = onboarding.getUsers();
+        users.forEach(user -> {
+            AddUserRoleDto userRoleDto = userMapper.toUserRole(onboarding);
+            userRoleDto.setProduct(productMapper.toProduct(onboarding, user));
+            Optional<Token> optToken = tokenRepository.findByOnboardingId(onboarding.getId());
+            userRoleDto.getProduct().setTokenId(optToken.map(Token::getId).orElse(null));
```

Comment:
you could avoid to retrieve token and setTokenId using `onboardingId` (there are the same value)